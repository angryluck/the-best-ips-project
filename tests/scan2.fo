fun int incr(int a, int b) = a + b
fun int writeInt(int n) = write(n)

fun [int] scan_case([int] a) =
  let b = scan(incr, 0, a) in
  map(writeInt, b)

fun [int] main() =
  let tmp = scan_case({1, 2, 3}) in
  let tmp = scan_case({10, 20}) in
  let tmp = scan_case({5}) in
  let tmp = scan_case({4, ~4, 4, ~4}) in
  let tmp = scan_case({100, 200, 300}) in
  let tmp = scan_case({0, 0, 0}) in
  let tmp = scan_case({1, 1, 1, 1, 1}) in
  let tmp = scan_case({2, 4, 6, 8}) in
  scan_case({9, ~3, 3})
