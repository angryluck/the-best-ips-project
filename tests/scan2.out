0 
1 
3
0
10
0
0
4
0
4
0
100
300
0
0
0
0
1
2
3
4
0
2
6
12
0
9
6
